chia/full_node/full_node.py

```diff
@@ -2223,8 +2227,12 @@ async def add_block(
 
         record = self.blockchain.block_record(block.header_hash)
         if self.weight_proof_handler is not None and record.sub_epoch_summary_included is not None:
-            if self._segment_task is None or self._segment_task.done():
-                self._segment_task = asyncio.create_task(self.weight_proof_handler.create_prev_sub_epoch_segments())
+            self._segment_task_list.append(
+                asyncio.create_task(self.weight_proof_handler.create_prev_sub_epoch_segments())
+            )
+            for task in self._segment_task_list[:]:
+                if task.done():
+                    self._segment_task_list.remove(task)
```

Comment:
`.remove()` is O(n) so doing it multiple times isn't great and often it's good to just recreate the filtered list as a single O(n) operation.

```suggestion
            self._segment_task_list = [
            	*(task for task in self._segment_task_list if not task.done()),
            	asyncio.create_task(self.weight_proof_handler.create_prev_sub_epoch_segments()),
            )
```

Reply:
this list is length n = 2 at most